### ISSUE TITLE
warn against incremental installs like `npm install foo` when foo isn't currently present

### ISSUE THREAD

**Issue by chenglou** [jordwalke/esy#40]
This breaks and is a very common pattern in js land. We should warn/prevent this here to avoid reinstall.

**Comment by jordwalke** [comment]
Why would this be an unsupported use case?

**Comment by chenglou** [comment]
(Following discord discussion) Didn't know it was only true for `npm install reason` in the current setup. Though @andreypopp did say something about a general `npm install foo` too.

**Comment by jordwalke** [comment]
With the current setup (and likely esy until we test) if you do npm install x, when you're *not already* in a `package.json` project, then it fails.

**Comment by chenglou** [comment]
Yes, that's what I meant

**Comment by andreypopp** [comment]
Yes, user won't see `foo` available in env unless they add it to `package.json`. We can emit a warning in case of "orphaned" packages in `node_modules/` dir.

**Comment by andreypopp** [comment]
Right now we have `esy add PACKAGENAME` command to add a new package. Should we consider this solved, @chenglou?

**Comment by chenglou** [comment]
Will I ever need to do `npm install foo`? What about in an existing project with mixed js and ocaml deps?

**Comment by andreypopp** [comment]
@chenglou with esy you use `esy install/add` to manage dependencies from npm (and npm compat sources like git repos). You don't need to use `npm install`.

**Comment by chenglou** [comment]
To be clear, `esy install babel` works as intended?
While I'm at it, is `esy build` a special command or can we plug e.g. webpack-related build steps with it?

**Comment by andreypopp** [comment]
> To be clear, esy install babel works as intended?

Yes.

> While I'm at it, is esy build a special command or can we plug e.g. webpack-related build steps with it?

`esy buiild` executes whatever configured in a corresponding package's `package.json` under `esy.build` key in a correctly set environment (see pjc spec for more details). That means you can put `webpack` somewhere in there. What's your use case?

**Comment by chenglou** [comment]
Just making sure that interop works and that Esy doesn't need to overhaul the entire setup and structure of an existing JS app.

As long as we document `esy install foo` and make it clear that `npm install foo` isn't recommended then I'm good!
